ega-data-api-res/src/main/java/eu/elixir/ega/ebi/reencryptionmvc/cache2k/My2KCachePageFactory.java

```diff
@@ -242,27 +207,13 @@ private CachePage loadPage(String key) {
             byte[] newIV = new byte[16]; // IV always 16 bytes long
             System.arraycopy(header.getIV(), 0, newIV, 0, 16); // preserved start value
             if (startCoordinate > 0) byteIncrementFast(newIV, startCoordinate);
-            decrypted = decrypt(buffer, sourceKey, newIV);
-        } catch (UnsupportedOperationException th) {
-            log.error("HTTP GET ERROR -1 " + th.toString() + "   -- " + byteRange + "\n" + url);
-            throw new ServerErrorException("Error Loading Cache Page -1 " + th.toString() + " for ", key);
-        } catch (IOException th) {
-            log.error("HTTP GET ERROR 0 " + th.toString() + "   -- " + byteRange + "\n" + url);
-            throw new ServerErrorException("Error Loading Cache Page 0 " + th.toString() + " for ", key);
-        } catch (IllegalBlockSizeException ex) {
-            log.error("HTTP GET ERROR 1 " + ex.toString() + "   -- " + byteRange + "\n" + url);
-            throw new ServerErrorException("Error Loading Cache Page 1 " + ex.toString() + " for ", key);
-        } catch (BadPaddingException ex) {
-            log.error("HTTP GET ERROR 2 " + ex.toString() + "   -- " + byteRange + "\n" + url);
-            throw new ServerErrorException("Error Loading Cache Page 2 " + ex.toString() + " for ", key);
+            return decrypt(buffer, sourceKey, newIV);
         } catch (Exception ex) {
-            log.error("HTTP GET ERROR 3 " + ex.toString() + "   -- " + byteRange + "\n" + url);
-            throw new ServerErrorException("Error Loading Cache Page 3 " + ex.toString() + " for ", key);
+            log.error("Error loading page " + ex.toString() + "   -- " + byteRange + "\n" + url);
```

Comment:
And please wrote a significant name for the exception that everybody can understand

Reply:
Could you pls elaborate more on how do you want it?

Reply:
This I read it the other way around so this is OK like it is :-)